gpg-interface.c

```diff
@@ -1035,12 +1038,23 @@ static int sign_buffer_ssh(struct strbuf *buffer, struct strbuf *signature,
 		goto out;
 	}
 
-	strvec_pushl(&signer.args, use_format->program,
-		     "-Y", "sign",
-		     "-n", "git",
-		     "-f", ssh_signing_key_file,
-		     buffer_file->filename.buf,
-		     NULL);
+	if (literal_ssh_key) {
+	    strvec_pushl(&signer.args, use_format->program,
+			"-Y", "sign",
+			"-n", "git",
+			"-U",
+			"-f", ssh_signing_key_file,
+			buffer_file->filename.buf,
+			NULL);
+	}
+	else {
+	    strvec_pushl(&signer.args, use_format->program,
+			"-Y", "sign",
+			"-n", "git",
+			"-f", ssh_signing_key_file,
+			buffer_file->filename.buf,
+			NULL);
+	}
```

Comment:
Rather than duplicating everything, it would be nicer to do

    strvec_pushl(&signer.args, /* options */, NULL);
    if (literal_ssh_key)
        strvec_push(&signer.args, "-U");
    strvec_push(buffer_file->filename.buf);

Reply:
Corrected